src/components/DxpFacilitySwitcher.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <ion-card>
+    <ion-card-header>
+      <ion-card-title>
+        {{ $t('Facility') }}
+      </ion-card-title>
+    </ion-card-header>
+    <ion-card-content>
+      {{ $t('Specify which facility you want to operate from. Order, inventory and other configuration data will be specific to the facility you select.') }}
+    </ion-card-content>
+    <ion-item lines="none">
+      <ion-label>{{ $t('Select facility') }}</ion-label>
+      <ion-select interface="popover" :value="userAppState.currentFacility.facilityId" @ionChange="setFacility($event)">
+        <ion-select-option v-for="facility in (userAppState.userProfile ? userAppState.userProfile.facilities : [])" :key="facility.facilityId" :value="facility.facilityId" >{{ facility.facilityName ? facility.facilityName : facility.name }}</ion-select-option>
+      </ion-select>
+    </ion-item>
+  </ion-card>
+</template>
+
+<script setup lang="ts">
+import { 
+  IonCard,
+  IonCardContent,
+  IonCardHeader,
+  IonCardTitle,
+  IonItem,
+  IonLabel,
+  IonSelect,
+  IonSelectOption
+} from '@ionic/vue';
+import { appContext } from '../index';
+import { computed } from 'vue';
+
+const emit = defineEmits(['check-facility', 'update-facility'])
+const appState = appContext.config.globalProperties.$store;
+
+const userAppState = computed(() => {
+  return {
+    currentFacility: appState.getters['user/getCurrentFacility'],
+    userProfile: appState.getters['user/getUserProfile'],
+    uploadProducts: appState.getters['product/getUploadProducts']
+  }
+});
+
+const setFacility = async (event: any) => {
+  const currentUserAppState = JSON.parse(JSON.stringify(userAppState.value))
+  const selectedFacility = event['detail'].value
+
+  if(currentUserAppState.currentFacility.facilityId && currentUserAppState.currentFacility.facilityId != selectedFacility && currentUserAppState.userProfile?.facilities) {
+    // check-facility is emitted before setFacility action.
+    emit('check-facility', selectedFacility)
+
+    await appState.dispatch('user/setFacility', {
+      'facility': currentUserAppState.userProfile.facilities.find((facility: any) => facility.facilityId == selectedFacility)
```

Comment:
While implementing in apps, check if this is done in the actions already, and remove it if so.